### DISCUSSION TITLE
API Rate Limits - documentation?

### DISCUSSION THREAD

**Discussion by jandy145** [ercot/api-specs#44] [Q&A] [answered]
I'm trying to extract historical data from the API (60 day SCED reports, 60 day DAM reports, and realtime SPPs). I run into rate limits (429 status code) on each API fairly quickly - after about 10 or 20 requests (sent without delay in between).

I want to make sure I'm using the API properly - are per-endpoint rate limits published anywhere? I wasn't able to find documentation.

**Comment by mackermann-ercot** [comment, marked as answer]
Hello there Jandy, 

Our current API request rate limit is approximately 2 requests per second (technically 30 requests per 60 seconds).  This rate has been set to reasonably balance performance and availability across the several million requests that are received each day.  If you have a specific need to request information at higher rate, please let us know and we will take it under consideration during the next (enhancement) planning phase.

Also, I will put in a feature request to have the rate limit documented within the user guide.

Regards,
-mackermann, ERCOT

**Reply by jandy145** [reply]
Thank you! No specific needs for now, but I will reach back out if that changes. Thanks for the info & looking forward to seeing the updated docs.

**Reply by Davidxswang** [reply]
> Hello there Jandy,
> 
> Our current API request rate limit is approximately 2 requests per second (technically 30 requests per 60 seconds). This rate has been set to reasonably balance performance and availability across the several million requests that are received each day. If you have a specific need to request information at higher rate, please let us know and we will take it under consideration during the next (enhancement) planning phase.
> 
> Also, I will put in a feature request to have the rate limit documented within the user guide.
> 
> Regards, -mackermann, ERCOT

Thanks for the info! I was looking for this limit in the doc as well, but I figured this out through the error 429. One small clarification is that "2 requests per second" is 120 (not 30) requests per 60 seconds. I guess you meant 0.5 requests per second.

**Reply by mackermann-ercot** [reply]
David, 

You are absolutely correct.  Apologies for the mistake. It is indeed 30 calls per 60 seconds which would be 1 request every 2 seconds, not 2 requests per second.  

Also, the request to provide this information within the documentation is on the upcoming list of features/enhancements.  

Regards,
-mackermann, ERCOT
